libraries/abstractions/common_io/test/test_iot_power.c

```diff
@@ -368,12 +375,12 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_EnterIdle )
     if( lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED )
     {
         TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    }
 
-    /* Read the performance mode and verify high performance mode */
-    lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
-    TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    TEST_ASSERT_EQUAL( eHighPowerPerfMode, xPowerMode );
+        /* Read the performance mode and verify high performance mode */
```

Comment:
This should be in else condition if you are fixing it. As if set_mode returns NOT_SUPPORTED, then we dont need to check for the mode.

Reply:
I'm not sure I understand..
If `set_mode` returns NOT_SUPPORTED as you mentioned.
Then, `lRetVal` will be equal to `NOT_SUPPORTED`.

This `if` condition checks if `lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED` before executing the code.
In this scenario, where high performance mode is not supported, we won't check for the mode.

Reply:
nm. You are correct.